routers/utils/utils.go

```diff
@@ -15,3 +18,57 @@ func RemoveUsernameParameterSuffix(name string) string {
 	}
 	return name
 }
+
+// ParseRepoOrderByType returns the OrderByType represented by the given string.
+func ParseRepoOrderByType(s string) models.RepoOrderByType {
+	switch s {
+	case "alphabetically":
+		return models.RepoOrderByAlphabetically
+	case "reversealphabetically":
+		return models.RepoOrderByAlphabeticallyReverse
+	case "leastupdate":
+		return models.RepoOrderByLeastUpdated
+	case "recentupdate":
+		return models.RepoOrderByRecentUpdated
+	case "newest":
+		return models.RepoOrderByNewest
+	case "oldest":
+		return models.RepoOrderByOldest
+	case "size":
+		return models.RepoOrderBySize
+	case "reversesize":
+		return models.RepoOrderBySizeReverse
+	case "id":
+		return models.RepoOrderByID
+	default:
+		return models.RepoOrderByRecentUpdated
+	}
+}
+
+// ToQueryString returns the string equivalent of the given RepoOrderByType, for
+// use in URL queries and HTML templates.
+func ToQueryString(s models.RepoOrderByType) string {
```

Comment:
As a method of `RepoOrderByType` is better.

Reply:
see https://github.com/go-gitea/gitea/pull/3116#discussion_r155997960